src/dataflow-types/src/client/controller/storage.rs

```diff
@@ -425,18 +426,31 @@ where
         &mut self,
         commands: Vec<(GlobalId, Vec<Update<Self::Timestamp>>, Self::Timestamp)>,
     ) -> Result<(), StorageError> {
-        for (id, updates, upper) in commands.iter() {
-            for update in updates {
-                if !update.timestamp.less_than(upper) {
+        for (id, updates, upper) in commands {
+            for update in &updates {
+                if !update.timestamp.less_than(&upper) {
                     return Err(StorageError::UpdateBeyondUpper(*id));
                 }
             }
+            let handles = self
+                .state
+                .persist_handles
+                .get_mut(&id)
+                .expect("unknown collection id");
+
+            let updates = updates
+                .into_iter()
+                .map(|u| (((), Ok(u.row)), u.timestamp, u.diff));
+            let timeout = Duration::from_secs(60);
+            handles
+                .write
+                .append(timeout, updates, Antichain::from_elem(upper))
```

Comment:
doesn't this want to be compare_and_append for tables?

Reply:
yes, fixed!